fetch.bs

```diff
@@ -3402,6 +3395,13 @@ optional <i>CORS flag</i> and <i>CORS-preflight flag</i>, run these steps:
       </ul>
 
       <p>then return a <a>network error</a>.
+
+     <li>
+      <p>If <var>actualResponse</var>'s <a for=response>URL</a>'s <a for=url>fragment</a> is null,
+      then set <var>actualResponse</var>'s <a for=response>URL</a>'s <a for=url>fragment</a> to
+      <var>request</var>'s <a for=request>current URL</a>'s <a for=url>fragment</a>.
+
+      <p class=note>Without this step, the fragment would end up getting lost due to the Cache API.
```

Comment:
In particular, a scenario @wanderview pointed out that I'm worried about is that `demo.svg` gets stored in the Cache API. A request for `demo.svg#viewBox...` comes in, `demo.svg` gets returned and the fragment identifier is lost.

This does suggest you have to be quite careful with how you store things in the cache if we keep the fragment around, or perhaps the Cache API should always drop the fragment beforehand?

@jakearchibald @jungkees @wanderview it'd be good to resolve that question and create a PR for the outcome against service workers.

Reply:
How does http cache handle fragments?  I assume it ignores fragments in terms of its matching?  I think it would be reasonable to match that behavior (assuming that is what it does...).

Reply:
I'm pretty sure that's what it does, yes. @mnot can confirm.

Reply:
Cache API stores a response without stripping its fragment. It ignores fragments in terms of matching. So, if a newly fetched response has a different fragment than the one in Cache, that new response replace the old record.

Shouldn't we serve the fragment as-is in the response's url even if its served from the cache? Response's url should always win even if fetch didn't send out the request to the network. Using a request's url's fragment for cached responses seems odd to me.

HTTP Cache doesn't store fragments?